ansible_rulebook/event_source/webhook.py

```
@@ -206,6 +369,28 @@ async def hmac_verify(
+
+    if "Authorization" not in request.headers:
+        raise web.HTTPUnauthorized()
+
+    parts = request.headers["Authorization"].strip().split(None, 1)
+    if len(parts) != 2 or parts[0] != "Bearer":
+        raise web.HTTPUnauthorized()
+
+    try:
+        await request.app[oauth_instance_key].authenticate(parts[1])
+    except AuthenticationFailed:
+        raise web.HTTPUnauthorized()
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)